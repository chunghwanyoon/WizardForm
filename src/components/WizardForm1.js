import React, { useRef, useEffect } from "react";
import styled, { keyframes } from "styled-components";

const leftToRight = keyframes`
  0% {
     margin-left: 30%;
  }
  80% {
    margin-left: 5%;
  }
  100% {
    margin-left: 0%;
  }
`;
const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  animation: ${leftToRight} 1s linear;
`;

export const FormContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: 2rem;
`;

const WizardForm1 = ({
  amount,
  backload_term,
  returning_term,
  nextStep,
  LoaneeInfo
}) => {
  const amountRef = useRef(amount);
  const backloadRef = useRef(backload_term);
  const returningRef = useRef(returning_term);
  console.log(amount);

  const validation = async () => {
    if (!amountRef.current.value) {
      amountRef.current.focus()
    } else if (!backloadRef.current.value ||  backloadRef.current.value.length < 1 ) {
      backloadRef.current.focus()
    } else if (!returningRef.current.value ||  returningRef.current.value.length < 1) {
      returningRef.current.focus()
    } else {
      await LoaneeInfo(
        amountRef.current.value,
        backloadRef.current.value,
        returningRef.current.value
      );
      nextStep();
    }
  }

  return (
    <Container>
      <FormContainer>
        <label htmlFor="amount">Amount</label>
        <input
          id="amount"
          type="text"
          placeholder="Enter amount"
          ref={amountRef}
        />
        <label htmlFor="backload">backload</label>
        <input
          id="backload"
          type="number"
          placeholder="Enter backload term"
          ref={backloadRef}
        />
        <label htmlFor="returning">Returning</label>
        <input
          id="returning"
          type="number"
          placeholder="Enter returning term"
          ref={returningRef}
        />
      </FormContainer>
      <div>
        <button>Prev</button>
        <button
          onClick={() => {
            console.log(amountRef.current.value);
            validation(
              amountRef.current.value,
              backloadRef.current.value,
              returningRef.current.value
            );
          }}
        >
          Next
        </button>
      </div>
    </Container>
  );
};

export default WizardForm1;
