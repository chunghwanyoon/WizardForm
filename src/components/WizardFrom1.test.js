import React from "react";
import { mount } from 'enzyme'
import WizardForm1 from './WizardForm1';

describe('WizardForm1', () => {
    it('match to initial snapshot', () => {
        const wrapper = mount(<WizardForm1></WizardForm1>);
        expect(wrapper).toMatchSnapshot();
    })

    it('renders props', () => {
        const wrapper = mount(<WizardForm1></WizardForm1>);
        expect(wrapper.props().amount).toBe(undefined)
    })
    
    // it('check the values', () => {
    //     const wrapper = mount(<WizardForm1></WizardForm1>);
    //     wrapper.find('input').simulate('change', {target: {  value: '200000'}});
    //     wrapper.update();
    //     expect(wrapper.find('input')).toBe('200000');
    // })
})