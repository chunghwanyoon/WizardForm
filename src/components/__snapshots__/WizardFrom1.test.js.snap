// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardForm1 match to initial snapshot 1`] = `
<WizardForm1>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "goXslx",
            "rules": Array [
              "
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  animation: ",
              Keyframes {
                "id": "sc-keyframes-iPYmJP",
                "inject": [Function],
                "name": "iPYmJP",
                "rules": Array [
                  "@-webkit-keyframes iPYmJP{0%{margin-left:30%;}80%{margin-left:5%;}100%{margin-left:0%;}}",
                  "@keyframes iPYmJP{0%{margin-left:30%;}80%{margin-left:5%;}100%{margin-left:0%;}}",
                ],
                "toString": [Function],
              },
              " 1s linear;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa goXslx"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "jLmMHk",
                  "rules": Array [
                    "
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: 2rem;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH jLmMHk"
            >
              <label
                htmlFor="amount"
              >
                Amount
              </label>
              <input
                id="amount"
                placeholder="Enter amount"
                type="text"
              />
              <label
                htmlFor="backload"
              >
                backload
              </label>
              <input
                id="backload"
                placeholder="Enter backload term"
                type="number"
              />
              <label
                htmlFor="returning"
              >
                Returning
              </label>
              <input
                id="returning"
                placeholder="Enter returning term"
                type="number"
              />
            </div>
          </StyledComponent>
        </styled.div>
        <div>
          <button>
            Prev
          </button>
          <button
            onClick={[Function]}
          >
            Next
          </button>
        </div>
      </div>
    </StyledComponent>
  </styled.div>
</WizardForm1>
`;
